### ISSUE TITLE
kotlin_library dependency on java_library not working?

### ISSUE THREAD

**Issue by guyshefner** [pubref/rules_kotlin#7]
Thanks for sharing your kotlin rules!

I'm trying to use a `java_library` from a `kotlin_library`, but it doesn't seem to be working for me.

Steps to reproduce the issue using the hello world example:
1. Clone kotlin_rules repo at commit 2e52fb9359298aa1c3d89248ceb6eef998fe99b0.
2. Build all works initially:
```
$ bazel build --strategy=KotlinCompile=standalone //examples/helloworld:all
```
3. Add the following `jars` line to `examples/helloworld/BUILD`:
```
kotlin_library(
    name = "rules",
    srcs = ["rules.kt"],
    jars = [":milk"]  # Added this line.
)
```
4. Now build all fails:
```
$ bazel build --strategy=KotlinCompile=standalone //examples/helloworld:all
ERROR: /tmp/rules_kotlin_clone/examples/helloworld/BUILD:25:1: in jars attribute of java_import rule //examples/helloworld:rules_kt: should not refer to Java rules. Since this rule was created by the macro 'kotlin_library', the error might have been caused by the macro implementation in /tmp/rules_kotlin_clone/kotlin/rules.bzl:200:16.
ERROR: Analysis of target '//examples/helloworld:rules_kt' failed; build aborted.
```

I'm using bazel version 0.4.4.

**Comment by guyshefner** [comment]
Here's another observation that strikes me as odd, but perhaps it's a useful clue: With the added `jars` line in step 3 of my previous comment, build `:all` fails to build `:rules`. But building `:rules` directly seems to work:
```
$ bazel build --strategy=KotlinCompile=standalone //examples/helloworld:rules
```

**Comment by pcj** [comment]
Thanks for reporting @guyshefner, I'll take a look and try to repro

**Comment by pcj** [comment]
The entry in the `jars` attribute needs to provide the actual name of the generated jar of the `java_library` rule:  For a `java_library` having name `:foo`, the name of the jar is `libfoo.jar`.  Therefore, the `kotlin_library` rule should read:

```python
kotlin_library(
    name = "rules",
    srcs = ["rules.kt"],
    jars = [":libmilk.jar"]  
)
```

Would be good to improve the error message somehow and certainly improve the documentation of `rules_kotlin` to avoid this in the future, because it's ~~definitely confusing~~ incorrect.

Please re-open if this does not address the problem.

**Comment by pcj** [comment]
@guyshefner I merged #8 and release v0.2.1 that addresses some of the issues above.  Pls report any issues with the new `java_deps`.  Thx!

**Comment by guyshefner** [comment]
The new java_deps is working for me. Thanks!
